src/ABLForcing/drivingForce/drivingForce.C

```diff
@@ -642,6 +688,36 @@ void Foam::drivingForce<Type>::findSingleForcingHeight_()
 }
 
 
+template<class Type>
+void Foam::drivingForce<Type>::findBlendLevels_()
+{
+    hLevelBlend0 = -1;
+    hLevelBlend1 = -1;
+    hLevelBlendMax = -1;
+
+    // Find the two levels closest to the specified height
+    // Find the closest level
+    forAllPlanes(zPlanes_,planeI)
+    {
+        if ( (hLevelBlend1 < 0) &&
+             (zPlanes_.planeLocationValues()[planeI] >= assimMaxHeight_) )
+        {
+            hLevelBlend1 = planeI;
+            hLevelBlend0 = planeI - 1;
```

Comment:
If assimMaxHeight_ is lower than the second grid level (e.g., naive user sets this height to 0), hLevelBlend0 will be negative. In the blending function you will then use negative indices in arrays, and I think this will result in a segmentation fault.

Another way for failure is when assimMaxHeight+blendThickness is higher than the last grid level.

I think it is best to add a safety catch to make sure that all hLevelBlend indices are positive to avoid segmentation errors.